src/main/java/org/candlepin/subscriptions/files/ProductWhitelist.java

```diff
@@ -42,7 +42,7 @@
     private final PerLineFileSource source;
 
     public ProductWhitelist(ApplicationProperties properties, ApplicationClock clock) {
-        if (properties.getProductWhitelistResourceLocation() != null) {
+        if (!properties.getProductWhitelistResourceLocation().isEmpty()) {
```

Comment:
Won't this throw a NullPointerException if not set? Perhaps we should check the null case as well?

Reply:
Fair enough, switched to `StringUtils.hasText` accordingly.